mlrun/package/packagers/pandas_packagers.py

```diff
@@ -94,27 +193,452 @@ class _CSVFormatter(_Formatter):
+            # Unflatten the dataframe:
+            obj = cls._unflatten_dataframe(dataframe=obj, **unflatten_kwargs)
+
+        return obj
+
+
+class _XLSXFormatter(_Formatter):
+    """
+    A static class for managing pandas xlsx files.
+    """
+
+    @classmethod
+    def to(
+        cls, obj: pd.DataFrame, file_path: str, flatten: bool = True, **to_kwargs
+    ) -> dict:
+        """
+        Save the given dataframe to the xlsx file path given.
+
+        :param obj:       The dataframe to save.
+        :param file_path: The file to save to.
+        :param flatten:   Whether to flatten the dataframe before saving. For some formats it is mandatory to enable
+                          flattening, otherwise saving and loading the dataframe will cause unexpected behavior
+                          especially in case it is multi-level or multi-index. Default to True.
+        :param to_kwargs: Additional keyword arguments to pass to the `to_excel` function.
+        """
+        # Get the engine to pass when unpacked:
+        instructions = {"engine": to_kwargs.get("engine", None)}
+
+        # Flatten the dataframe (this format have problems saving multi-level dataframes):
+        if flatten:
+            obj, unflatten_kwargs = cls._flatten_dataframe(dataframe=obj)
+            instructions["unflatten_kwargs"] = unflatten_kwargs
+
+        # Write to xlsx:
+        obj.to_excel(excel_writer=file_path, **to_kwargs)
+
+        return instructions
+
+    @classmethod
+    def read(
+        cls, file_path: str, unflatten_kwargs: dict = None, **read_kwargs
+    ) -> pd.DataFrame:
         """
-        Read the dataframe / series from the given parquet file path.
+        Read the dataframe from the given xlsx file path.
 
-        :param file_path:   The file to read the dataframe from.
-        :param read_kwargs: Additional keyword arguments to pass to the `read_csv` function.
+        :param file_path:        The file to read the dataframe from.
+        :param unflatten_kwargs: Unflatten keyword arguments for unflattening the read dataframe.
+        :param read_kwargs:      Additional keyword arguments to pass to the `read_excel` function.
 
-        :return: The loaded dataframe / series.
+        :return: The loaded dataframe.
         """
-        return pd.read_csv(filepath_or_buffer=file_path, **read_kwargs)
+        # Read the xlsx:
+        obj = pd.read_excel(io=file_path, **read_kwargs)
+
+        # Check if it was flattened in packing:
+        if unflatten_kwargs is not None:
+            # Remove the default index (joined with reset index):
+            if obj.columns[0] == "Unnamed: 0":
+                obj.drop(columns=["Unnamed: 0"], inplace=True)
+            # Unflatten the dataframe:
+            obj = cls._unflatten_dataframe(dataframe=obj, **unflatten_kwargs)
+
+        return obj
+
+
+class _HTMLFormatter(_Formatter):
+    """
+    A static class for managing pandas html files.
+    """
+
+    @classmethod
```

Comment:
I think the `to` and `read` functions should be staticmethods instead of classmethods
```
When to use the class or static method?
We generally use the class method to create factory methods. Factory methods return class objects ( similar to a constructor ) for different use cases.
We generally use static methods to create utility functions.
```
classmethods can also be admissible when you access/change the class state which is not the case here.
here we return a dataframe and not the actual class of the method
from [here](https://www.geeksforgeeks.org/class-method-vs-static-method-python/)

Reply:
I'll go with you :+1:

Reply:
Keep in mind the author forgot an important usage: `classmethods` are also used for inheritance so a method will access the correct method / class argument. The whole `_Formatter` and `_Archiver` classes are built the same with `classmethod`s because some use this feature (inheritance) but like here, some don't. So, for consistency I'd rather keep these as class methods.